### PR TITLE
docs: Add documentation to use Typescript with Vue 2/3 on Write Stories page

### PR DESCRIPTION
**Pull request by valentinpalkovic**
## What I did

Added documentation of using Vue 2/3 with Typescript on the "Write Stories" documentation page

### CODE REVIEW

File: docs/writing-stories/introduction.md
Comment (jonniebigodes) [inline docs/writing-stories/introduction.md:150]: @valentinpalkovic if you could rename the files and update the references in the documentation to for instance: "vue/button-story-using-args.ts-2.ts.mdx" for Vue 2 and "vue/button-story-using-args.ts-3.ts.mdx" for Vue 3,   I'd appreciate it. The reason is that if you spin up the site with the file naming you've used it will yield the following issue with the documentation:

files-render-issue

Renaming them, it will render them correctly and select the correct snippet, as you can see below:

files-render-ok

Let me know once you've updated it and we'll go from there.

File: docs/snippets/vue/button-group-story.3.ts.mdx
Comment (jonniebigodes) [inline docs/snippets/vue/button-group-story.3.ts.mdx:26]: @valentinpalkovic one thing I've noticed with the snippets is that the types ComponentMeta and ComponentStory don't exist in Vue 2 and Vue 3. I was checking this and adding a story with those types yields the following error:

vue-3-type-errors

But both `Meta` and `Story` types do exist and it seems that you could use them as such, for example with Vue 3:

```ts
import Button from "./Button.vue";

import { Meta, Story } from "@storybook/vue3";

// types don't exist in TS for Vue 3
// import { ComponentMeta, ComponentStory } from "@storybook/vue3";

//👇 This default export determines where your story goes in the story list
export default {
  title: "MyButton",
  component: Button,
  argTypes: {
    backgroundColor: { control: "color" },
    onClick: {},
    size: {
      control: { type: "select" },
      options: ["small", "medium", "large"],
    },
  },
} as Meta;

//👇 We create a “template” of how args map to rendering
const Template: Story = (args: unknown) => ({
  components: { Button },
  setup() {
    //👇 The args will now be passed down to the template
    return { args };
  },
  template: '<Button v-bind="args" />',
});
```

And for Vue 2:
```ts
import Button from "./Button.vue";

import { Meta, Story } from "@storybook/vue";

export default {
  title: "Button",
  component: Button,
  argTypes: {
    backgroundColor: { control: "color" },
    size: {
      control: { type: "select" },
      options: ["small", "medium", "large"],
    },
  },
} as Meta;

const Template: Story = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { Button },
  template: '<Button @onClick="onClick" v-bind="$props" />',
});
```

@shilman are you ok with using this approach with TS+Vue based examples for now? And once and if the types are exported we can adjust the documentation accordingly. Would love to hear your feedback on this.

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@valentinpalkovic thanks for taking the time and effort into this pull request and helping us with the documentation. We're extremely thankful for this. There are a couple of items that need addressing and once they're solved I'd gladly merge this. 

One quick tip for you if you allow me to help you and the rest of the maintainers get the documentation pull requests merged faster. As you're going through making your changes, it would be best to check and see if there are any issues with the docs by previewing them locally. [Here](https://storybook.js.org/docs/react/contribute/new-snippets#preview-your-work) is some documentation that can help you get started while working with Storybook docs.


Hope you have a fantastic weekend!

Stay safe.

**Comment by valentinpalkovic** [comment]
> @valentinpalkovic thanks for taking the time and effort into this pull request and helping us with the documentation. We're extremely thankful for this. There are a couple of items that need addressing and once they're solved I'd gladly merge this.
> 
> One quick tip for you if you allow me to help you and the rest of the maintainers get the documentation pull requests merged faster. As you're going through making your changes, it would be best to check and see if there are any issues with the docs by previewing them locally. [Here](https://storybook.js.org/docs/react/contribute/new-snippets#preview-your-work) is some documentation that can help you get started while working with Storybook docs.
> 
> Hope you have a fantastic weekend!
> 
> Stay safe.

Great! I was missing this site (https://storybook.js.org/docs/react/contribute/new-snippets#preview-your-work) I was searching for a way to spin up the docs locally but I didn‘t find this documentation.

File: docs/writing-stories/introduction.md
Comment (valentinpalkovic) [reply docs/writing-stories/introduction.md]: @jonniebigodes Updated!

File: docs/snippets/vue/button-group-story.3.ts.mdx
Comment (valentinpalkovic) [reply docs/snippets/vue/button-group-story.3.ts.mdx]: You are totally right. I can't explain, why I have used the wrong types of imports.

@jonniebigodes I have corrected it a bit differently. Instead of using:

```ts
import { Meta, Story } from "@storybook/vue";
import { Meta, Story } from "@storybook/vue3";
```

I have used

```ts
import { Meta, StoryFn } from "@storybook/vue";
import { Meta, StoryFn } from "@storybook/vue3";
```

Because in Storybook v7 `Story` will be the new type for the [Component Story Format 3.0](https://storybook.js.org/blog/component-story-format-3-0/) and StoryFn stays the type for the CSF 2.0 format.

Comment (valentinpalkovic) [reply docs/snippets/vue/button-group-story.3.ts.mdx]: @jonniebigodes adjusted!

Comment (jonniebigodes) [reply docs/snippets/vue/button-group-story.3.ts.mdx]: Thanks for taking care of this, appreciate it 🙏 !

File: docs/writing-stories/introduction.md
Comment (jonniebigodes) [reply docs/writing-stories/introduction.md]: That's great, thanks for making the adjustments!

**Comment by jonniebigodes** [comment]
@valentinpalkovic thanks for taking the time to make the adjustments, appreciate it. Going to merge this! 

Looking forward to your next contribution!

Hope you have a great weekend!

Stay safe

**Pull request merged**
